From 46cd721c0ff2f319fbc523c0990a40648f109b62 Mon Sep 17 00:00:00 2001
From: Ander Juaristi <ajuaristi@gmx.es>
Date: Fri, 20 Nov 2015 17:35:31 +0000
Subject: Fix potential NULL pointer dereference

 * src/gnutls.c (ssl_connect_wget): check for NULL before calls
---
diff --git src/gnutls.c src/gnutls.c
index 828050b..d1444fe 100644
--- src/gnutls.c
+++ src/gnutls.c
@@ -633,9 +633,13 @@ ssl_connect_wget (int fd, const char *hostname, int *continue_session)
         {
           if (!ctx || !ctx->session_data || gnutls_session_set_data (session, ctx->session_data->data, ctx->session_data->size))
             {
-              /* server does not want to continue the session */
-              gnutls_free (ctx->session_data->data);
-              gnutls_free (ctx->session_data);
+              if (ctx && ctx->session_data)
+                {
+                  /* server does not want to continue the session */
+                  if (ctx->session_data->data)
+                    gnutls_free (ctx->session_data->data);
+                  gnutls_free (ctx->session_data);
+                }
               gnutls_deinit (session);
               return false;
             }
--
cgit v0.9.0.2
